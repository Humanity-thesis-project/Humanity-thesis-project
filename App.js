import React from 'react';
import conf from './config.js';
import { StyleSheet, Text, View, AppRegistry, Image, KeyboardAvoidingView, TouchableOpacity, Dimensions} from 'react-native';
import SignUp from './client/signUp';
import SignIn from './client/signIn';

import List from './client/list';
import Navbar from './client/navbar';
export default class App extends React.Component {
  constructor() {
    super();
    this.state = {
      showSignIn: false,
      showSignUp: false,
      showMain: true,
      isLogged: false,
      type:"",
      info: {}
    };
    const width = Dimensions.get('window').width
    const height = Dimensions.get('window').height
  }
  signin = () => {
    this.setState({ showSignUp: false});
    this.setState({ showSignIn: true});
    this.setState({ showMain: false });
  }; 
  signup = () => {
    this.setState({ showSignUp: true });
    this.setState({ showSignIn: false });
    this.setState({ showSMain: false });
  }

  // isLoggedIn () {
  //   fetch(conf.url + '/isLoggedIn',
  //     {method:'GET'})
  //   .then((response) => response.json())
  //     .then((data) => {
  //       console.log('------------------------------------>')
  //       console.log(data)
  //       this.state.isLogged = data.isLoggedIn;
  //       this.state.type = data.type
  //     })
  //     .catch((error) => {
  //       console.error(error);
  //    });
  // }

  main() {
 
  const mainComponent =  
        <View style={{marginTop: 30, marginLeft:80,marginRight:50}} >
        <Image source={require('./images/200x200-icon-drop.png')}/>
   
        <Text style={{fontSize:18,marginTop:20,fontStyle:'italic'}}>Welcome to our application 'Humanity' If you are from our family</Text>
        <Text style={{fontSize:18,marginTop:20,fontStyle:'italic'}}> and you want to signIn WELCOME from here press on signIn</Text>
      
        <TouchableOpacity onPress={this.signin.bind(this)}>

          <Text>Sign In</Text>
        </TouchableOpacity>
    <Text style={{fontSize:18,marginTop:20,fontStyle:'italic'}} > Or if you want to JOIN US WELCOME from here</Text>
        <TouchableOpacity onPress={this.signup.bind(this)}>

          <Text>Sign Up</Text>
        </TouchableOpacity>
     </View>

  if (this.state.showSignIn && !this.state.mainComp && !this.state.showSignUp) {
    return <SignIn/>
  }
  else if(!this.state.showSignIn && this.state.showMain && !this.state.showSignUp){
    return mainComponent;
     
  }
  else if(this.state.showSignUp && !this.state.mainComp && !this.state.showSignIn) {
    return <SignUp/>;
  }
}

  render() {
    return (
      <View>
       {this.main()}
      </View>
      
    );
  }
}


AppRegistry.registerComponent('App', () => App);



    
  
//    }
// });
