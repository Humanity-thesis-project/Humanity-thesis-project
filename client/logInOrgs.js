
import React from "react";
import { StyleSheet, Text, TextInput, View,TouchableOpacity, Button} from "react-native";
import OrgProfile from "./orgprofile";
import Navbar from "./navbar";

import conf from '../config.js';

export default class logInOrgs extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      name:'',
      password:'',
      signedIn: false,
      orgInfo: {}
    };
    fetch(conf.url + '/orgs/signout',
      {method:'GET'})
  }
    submitSignIn () { 
      
     fetch(conf.url + '/orgs/signin',
      {
          method:'POST',
          headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json',
          },
          body:JSON.stringify({
            name:this.state.name ,
            password:this.state.password})
        }).then((response) => response.json())
           .then((data) => {
            console.log('------------------------------------>')
            console.log(data.events) 
              this.state.orgInfo = data;
              this.setState({signedIn: true})
          })
            .catch((error) => {
                console.error(error);
            });
    }
  

goToProfile () {
  if(this.state.signedIn)
    return <Navbar info = {this.state.orgInfo} profile = {"org"} events = {this.state.orgInfo.events}/>;

  else{
    return <View style = {{marginTop:200,  alignItems: 'center' }}>
      <Text style={{fontWeight: "bold", textAlign: 'center', marginBottom: 10,fontSize:30}}> Sign In </Text>
     <Text style = {{marginRight:80}}>Orgenization name:</Text>
      <TextInput
          style={{height: 50, width: 200 ,alignItems: 'center'}}
          returnKeyType='next'
          placeholder="Enter Yuor Username"
          onChangeText={(name) => this.setState({name})}
          value={this.state.username}
        />
        <Text style = {{marginRight:130}}>Password:</Text>
        <TextInput
          returnKeyType='go'
          style={{height: 50, width: 200,alignItems: 'center'}}
          placeholder="Enter Your Password"
          secureTextEntry = {true}
          onChangeText={(password) => this.setState({password})}
          value={this.state.password}
        />
        <Button title = "submit" onPress = {this.submitSignIn.bind(this)}/>
      </View>
    }
}


    render() {
        return (
            <View>
                {this.goToProfile()}
            </View>
        );
    }
}
