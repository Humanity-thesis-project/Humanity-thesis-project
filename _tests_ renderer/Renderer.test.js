import 'react-native';
import React from 'react';
import App from '../App';

import Createevents from '../client/createevents';

import Entryevent from '../client/entryevent';

import List from '../client/list';

import logInOrgs from '../client/logInOrgs';

import logInUseres from '../client/logInUsers';

import UserSignUp from '../client/userSignUp';

import Navbar from '../client/navbar';

import renderer from 'react-test-renderer';
 	
describe('Client test',() =>{
	
  it('shows main component', () =>{
  	const tree = renderer.create(
    	<App />
  	).toJSON();

  	expect(tree).toMatchSnapshot();
  });

  it('shows list of events', () => {
  	const tree = renderer.create(
    	<List />
  	).toJSON();

  	expect(tree).toMatchSnapshot();
  });

  it('shows event informations', () => {
  	const tree = renderer.create(
    	<Entryevent />
  	).toJSON();

  	expect(tree).toMatchSnapshot();
  });

  it('after done shows org profile', () => {
  	const tree = renderer.create(
    	<logInOrgs />
  	).toJSON();

  	expect(tree).toMatchSnapshot();
  });

  it('after done shows events page', () => {
  	const tree = renderer.create(
    	<logInUseres />
  	).toJSON();

  	expect(tree).toMatchSnapshot();
  });

  it('shows sign up page for users', () => {
  	const tree = renderer.create(
    	<UserSignUp />
  	).toJSON();

  	expect(tree).toMatchSnapshot();
  });

  it('shows sign up page for users', () => {
    const tree = renderer.create(
      <Navbar />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });


});