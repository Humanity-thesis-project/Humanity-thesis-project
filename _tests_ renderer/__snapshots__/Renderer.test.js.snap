// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client test after done shows events page 1`] = `<logInUseres />`;

exports[`Client test after done shows org profile 1`] = `<logInOrgs />`;

exports[`Client test shows list of events 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Events
    </Text>
  </View>
</View>
`;

exports[`Client test shows main component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "gray",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Image
      source={1}
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Wlcom to our applecation ['Humanity'] If you are from our family
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
       and you want to signIn WELCOM from here press on signIn
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        Login
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
       Or if you want to JOIN US WELCOM from here
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        Sign up
      </Text>
    </View>
  </View>
</View>
`;

exports[`Client test shows sign up page for users 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
       Sign Up 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Username:
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter Username"
      returnKeyType="next"
      style={
        Object {
          "height": 50,
          "width": 200,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Email:
    </Text>
    <TextInput
      autoCapitalize="none"
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="Enter Email"
      returnKeyType="next"
      style={
        Object {
          "height": 50,
          "width": 200,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Password:
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter Password"
      returnKeyType="go"
      secureTextEntry={true}
      style={
        Object {
          "height": 50,
          "width": 200,
        }
      }
    />
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          submit
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Client test shows sign up page for users 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "gray",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "fontSize": 30,
        "fontWeight": "bold",
        "margin": 5,
        "opacity": 1,
        "textAlign": "center",
        "width": 50,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "width": 40,
        }
      }
    >
       options 
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
     click to show options 
  </Text>
</View>
`;
